### ISSUE TITLE
[BUG] Line draw mode does not work when drawing to a canvas

### ISSUE THREAD

**Issue by Nehemek** [lovebrew/LovePotion#137]
### **Describe the bug**
Drawing a shape inside a canvas with the "line" draw mode does not work, it will instead draw in filled mode.

### **Expected behavior**
"Line" draw mode should work properly on canvases.

### **Relevant Code To Reproduce**
```
function love.load()
    love.graphics.setBackgroundColor(0.8,0.75,1)
    love.graphics.set3D(false)
    canvas = love.graphics.newCanvas(200,240)
end

function love.draw(screen)
    if screen == "top" then
        love.graphics.setColor(1,1,1,1)
        love.graphics.circle("line",100,120,16)

        love.graphics.setCanvas(canvas)
        love.graphics.clear(255,255,255,0)
        love.graphics.setColor(1,1,1,1)
        love.graphics.circle("line",100,120,16)
        love.graphics.setCanvas()

        love.graphics.setColor(1,1,1,1)
        love.graphics.draw(canvas,200,0)
    end
end

function love.gamepadpressed(joystick, button)
    if button == "start" then love.event.quit() end
end
```

### **[Screenshots]**
Results of the above code:
![linemodebug](https://user-images.githubusercontent.com/66986051/116470391-fe1ed300-a838-11eb-9ff9-23061a0875b4.png)

### **System Info [Please complete the following information]:**
System: 3ds v11.14.0-46U
CFW: Luma v10.2.1
Method of execution: Homebrew App
LÖVE Potion version: 2.1.0 (development version)

**Comment by TurtleP** [comment]
Related to #135. Hopefully I can pinpoint what the root cause is and that should fix this too.

**Comment by TurtleP** [comment]
Fixed with commit 7bda06a
